src/main/java/org/codinjutsu/tools/jenkins/view/BuildParamDialog.java

```diff
@@ -117,7 +118,20 @@ private void addParameterInputs() {
         for (JobParameter jobParameter : parameters) {
             final JobParameterRenderer jobParameterRenderer = JobParameterRenderer.findRenderer(jobParameter)
                     .orElseGet(ErrorRenderer::new);
-            final JobParameterComponent<?> jobParameterComponent = jobParameterRenderer.render(jobParameter);
+            JobParameterComponent<?> jobParameterComponent;
+            if (jobParameter.getChoices().isEmpty() && jobParameterRenderer instanceof GitParameterRenderer) {
```

Comment:
Should this logioc not be added in `GitParameterRenderer`
e.g load values if not available

Reply:
I checked the code of `GitParameterRenderer `, how about adding this to its `render` method ?

Reply:
You can retreive a service for RequestManager and then load need ed data.

I try to edit your branch the days.

Reply:
ok.

Reply:
I changed the render pipeline to allow loading of additional Values for each renderer.
Could you please check if it still works for you

I also improved it with URI Encoding an dadd support an additional Git Parameter Plugin (ListGitBranchesParameter)

Reply:
I tested, it's ok. Thanks